nifi-nar-bundles/nifi-beats-bundle/nifi-beats-processors/src/main/java/org/apache/nifi/processors/beats/netty/BeatsMessageFactory.java

```diff
@@ -14,24 +14,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.nifi.processors.beats.event;
+package org.apache.nifi.processors.beats.netty;
 
-import org.apache.nifi.processor.util.listen.event.EventFactory;
-import org.apache.nifi.processor.util.listen.response.ChannelResponder;
+import org.apache.nifi.processor.util.listen.event.NetworkEventFactory;
+import org.apache.nifi.processors.beats.frame.BeatsMetadata;
 
 import java.util.Map;
 
 /**
- * An EventFactory implementation to create BeatEvents.
+ * An EventFactory implementation to create BeatsMessages.
  */
-public class BeatsEventFactory implements EventFactory<BeatsEvent> {
+public class BeatsMessageFactory implements NetworkEventFactory<BeatsMessage> {
 
     @Override
-    public BeatsEvent create(final byte[] data, final Map<String, String> metadata, final ChannelResponder responder) {
-        final String sender = metadata.get(EventFactory.SENDER_KEY);
-        final int seqNumber = Integer.valueOf(metadata.get(BeatsMetadata.SEQNUMBER_KEY));
-
-        return new BeatsEvent(sender, data, responder, seqNumber);
+    public BeatsMessage create(final byte[] data, final Map<String, String> metadata) {
+        final int txnr = Integer.valueOf(metadata.get(BeatsMetadata.SEQNUMBER_KEY));
```

Comment:
Recommend renaming this variable:
```suggestion
        final int sequenceNumber = Integer.valueOf(metadata.get(BeatsMetadata.SEQNUMBER_KEY));
```